### DISCUSSION TITLE
Disabling Specific Hotkeys

### DISCUSSION THREAD

**Discussion by mrsmiles** [pelicanmapping/osgearth#2395] [Support Q&A] [answered]
OSG Version: 3.4.0
OSGEarth Version: 2.7
Gfx Card: NVIDIA Quadro P4000

I am looking for a way to disable specific hotkey functionality. Specifically, the escape key closes the viewer window and I'd like it to not do that. The proper thing to do would probably be to intercept the keypress signal and kill it in my application, however this is not working for some reason -- OSGEarth still receives the signal and closes the viewer. I was wondering if there was an easy way I could disable that functionality (like a command line arg to pass in during compile) in OSGEarth, other than just completely removing the code and rebuilding. Any tips are appreciated, let me know if more info is needed.

**Comment by gwaldron** [comment, marked as answer]
I am not aware of a compile-time flag to disable it.
This functionality comes from OpenSceneGraph (not osgEarth); I believe you can call `osg::Viewer::setKeyEventSetsDone(0);` to disable close-on-ESC.

**Reply by mrsmiles** [reply]
Thank you for the quick response! I should have known this was OSG behavior, not OSGEarth. I have set the setDone key on my compositeViewer, but the closing behavior is not being prevented. I will have to dig into it some more. Thanks again!
